docs/pipelines.md

```diff
@@ -880,10 +900,119 @@ This kind of variable can have any one of the values from the following table:
 
 For an end-to-end example, see [`status` in a `PipelineRun`](../examples/v1beta1/pipelineruns/pipelinerun-task-execution-status.yaml).
 
+### Guard `Finally Task` execution using `WhenExpressions`
+
+Similar to `Tasks`, `Finally Tasks` can be guarded using [`WhenExpressions`](#guard-task-execution-using-whenexpressions)
+that operate on static inputs or variables. Like in `Tasks`, `WhenExpressions` in `Finally Tasks` can operate on
+`Parameters` and `Results`. Unlike in `Tasks`, `WhenExpressions` in `Finally Tasks` can also operate on the [`Execution
+Status`](#using-execution-status-of-pipelinetask) of `Tasks`.
+
+#### `WhenExpressions` using `Parameters` in `Finally Tasks`
+
+`WhenExpressions` in `Finally Tasks` can utilize `Parameters` as demonstrated using [`golang-build`](https://github.com/tektoncd/catalog/tree/master/task/golang-build/0.1)
+and [`send-to-channel-slack`](https://github.com/tektoncd/catalog/tree/master/task/send-to-channel-slack/0.1) Catalog
+`Tasks`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: pipelinerun-
+spec:
+  pipelineSpec:
+    params:
+    - name: enable-notifications
+      type: string
+      description: a boolean indicating whether the notifications should be sent
+    tasks:
+      - name: golang-build
+        taskRef:
+          name: golang-build
+      # […]
+    finally:
+      - name: notify-build-failure # executed only when build task fails and notifications are enabled
+        when:
+          - input: $(tasks.golang-build.status)
```

Comment:
I see most of the examples or tests refer to task status, is there a way to guard again pipeline run status? I think the user in the send to slack example would typically would want to grab the status of the execution of the whole pipeline if it has failed or succeeded, what do you think having this as the example?

Reply:
pipeline run status is calculated based on both tasks and finally tasks, as described [here](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#pipelinerun-status-with-finally), so I don't see how we can guard against that because it won't be available yet when executing finally tasks

if you're referring to the overall status of the tasks only, not finally tasks, I can imagine that a user could pass in all the statuses of the tasks to the values array, as shown below 

```yaml
finally:
      - name: notify-failure # executed only when tasks failed
        when:
          - input: "Failed"
 	    operator: in
            values: [$(tasks.unit-tests.status), $(tasks.golang-build.status), $(tasks.deploy.status)]         
```

Reply:
/cc @pritidesai on execution status variables available in finally

Reply:
@chmouel I have a WIP proposal to support [aggregate](https://github.com/tektoncd/pipeline/issues/1020#issuecomment-747156100) status of `tasks` section, would that work for your use case? 

The proposal includes introducing a new context variable `$(tasks.status)` (feedback on syntax is welcome) 🙏. `$(tasks.status)` would be set to `Failed` if any on of the `dag` task fails, and `Succeeded` if all `dag` tasks succeed.

Reply:
the aggregate status of `tasks` would be a very useful in `whenexpressions` in `finally tasks`, thanks for the update @pritidesai!